yb-voyager/cmd/import_data_test.go

```diff
@@ -496,6 +379,97 @@ INSERT INTO test_schema.test_data (name) VALUES ('name_%d');`, i))
+
+	// Start Postgres container.
+	postgresContainer := testcontainers.NewTestContainer("postgresql", nil)
+	if err := postgresContainer.Start(ctx); err != nil {
+		utils.ErrExit("Failed to start Postgres container: %v", err)
+	}
+
+	// Start YugabyteDB container.
+	yugabytedbContainer := testcontainers.NewTestContainer("yugabytedb", nil)
+	if err := yugabytedbContainer.Start(ctx); err != nil {
+		utils.ErrExit("Failed to start YugabyteDB container: %v", err)
+	}
+
+	createSchemaSQL := `CREATE SCHEMA IF NOT EXISTS test_schema;`
+	dropSchemaSQL := `DROP SCHEMA IF EXISTS test_schema CASCADE;`
+	createTableSQL := testutils.AllTypesSchemaDDL("test_schema.all_types")
+	insertDataSQL := testutils.AllTypesInsertSQL("test_schema.all_types", 10000)
+
+	postgresContainer.ExecuteSqls(createSchemaSQL, createTableSQL)
+	postgresContainer.ExecuteSqls(insertDataSQL)
+	defer postgresContainer.ExecuteSqls(dropSchemaSQL)
+
+	// create the same table in YugabyteDB without data.
+	yugabytedbContainer.ExecuteSqls(createSchemaSQL, createTableSQL)
+	defer yugabytedbContainer.ExecuteSqls(dropSchemaSQL)
+
+	_, err := testutils.RunVoyagerCommand(postgresContainer, "export data", []string{
+		"--export-dir", exportDir,
+		"--source-db-schema", "test_schema",
+		"--disable-pb", "true",
+		"--yes",
+	}, nil, false)
+	if err != nil {
+		t.Fatalf("Export command failed: %v", err)
+	}
+
+	importDataCmdArgs := []string{
+		"--export-dir", exportDir,
+		"--disable-pb", "true",
+		"--on-primary-key-conflict", "ERROR",
+		"--yes",
+	}
+
+	// first run: import data command to load data from PG
+	_, err = testutils.RunVoyagerCommand(yugabytedbContainer, "import data", importDataCmdArgs, nil, false)
+	if err != nil {
+		t.Fatalf("Import command failed: %v", err)
+	}
+
+	// second run: test IGNORE on primary key conflict
+	importDataCmdArgs = []string{
+		"--export-dir", exportDir,
+		"--disable-pb", "true",
+		"--on-primary-key-conflict", "IGNORE",
+		"--batch-size", "10",
+		"--start-clean", "true",
+		"--yes",
+	}
+	// second run: to test INSERT ON CONFLICT DO NOTHING statements with all datatypes
+	_, err = testutils.RunVoyagerCommand(yugabytedbContainer, "import data", importDataCmdArgs, nil, false)
+	if err != nil {
+		t.Fatalf("Import command failed: %v", err)
+	}
+
+	// Connect to both Postgres and YugabyteDB.
+	pgConn, err := postgresContainer.GetConnection()
+	if err != nil {
+		t.Fatalf("Error connecting to Postgres: %v", err)
+	}
+	ybConn, err := yugabytedbContainer.GetConnection()
+	if err != nil {
+		t.Fatalf("Error connecting to YugabyteDB: %v", err)
+	}
+
+	// Compare the full table data between Postgres and YugabyteDB.
+	// We assume the table "test_data" has a primary key "id" so we order by it.
+	if err := testutils.CompareTableData(ctx, pgConn, ybConn, "test_schema.all_types", "id"); err != nil {
+		t.Errorf("Table data mismatch between Postgres and YugabyteDB: %v", err)
+	} else {
+		t.Log("Success: Table data in Postgres and YugabyteDB match exactly.")
```

Comment:
a success log is pretty much useless, since the test passing is an indication of success.

Reply:
multiple parallel PRs have this. Will clean it for once, later.